src/ngraph/op/pad.cpp

```diff
@@ -305,23 +305,6 @@ void op::v1::Pad::validate_and_infer_types()
     const auto& pads_begin_coord = get_pads_begin();
     const auto& pads_end_coord = get_pads_end();
 
-    for (const auto& pads_begin_dim : pads_begin_coord)
```

Comment:
According to specification all elements should be non-negative. However, `Pad:v0` supports negative elements of `pad_begin` and `pad_end` (it causes removing elements from indicated axes). @ilyachur @GlebKazantaev  Can we remove this constraint from op?

Reply:
To be honest I'm having trouble remembering why we had this capability in `v0`.

Reply:
It might have been so an op with padding (which can be negative) could be decomposed into an unpadded op preceded by a pad op.